src/openms/include/OpenMS/FORMAT/AbsoluteQuantitationMethodFile.h

```diff
@@ -0,0 +1,102 @@
+// --------------------------------------------------------------------------
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL ANY OF THE AUTHORS OR THE CONTRIBUTING
+// INSTITUTIONS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;
+// OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+// WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR
+// OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
+// ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// --------------------------------------------------------------------------
+// $Maintainer: Douglas McCloskey $
+// $Authors: Douglas McCloskey $
+// --------------------------------------------------------------------------
+
+#ifndef OPENMS_FORMAT_ABSOLUTEQUANTITATIONMETHODFILE_H
+#define OPENMS_FORMAT_ABSOLUTEQUANTITATIONMETHODFILE_H
+
+#include <OpenMS/FORMAT/CsvFile.h>
+#include <OpenMS/CONCEPT/ProgressLogger.h>
+#include <OpenMS/DATASTRUCTURES/StringListUtils.h>
+#include <OpenMS/ANALYSIS/QUANTITATION/AbsoluteQuantitationMethod.h>
+
+namespace OpenMS
+{
+  /**
+      @brief File adapter for AbsoluteQuantitationMethod files.
+
+      loads and stores .csv or .tsv files describing an AbsoluteQuantitationMethod.
+
+      @ingroup FileIO
+  */
+  class OPENMS_DLLAPI AbsoluteQuantitationMethodFile :
+    private CsvFile,
+    public ProgressLogger
+  {
+public:
+    ///Default constructor
+    AbsoluteQuantitationMethodFile();
+    ///Destructor
+    virtual ~AbsoluteQuantitationMethodFile();
+
+    /**
+        @brief Loads an AbsoluteQuantitationMethod file.
+
+        @exception Exception::FileNotFound is thrown if the file could not be opened
+        @exception Exception::ParseError is thrown if an error occurs during parsing
+    */
+    void load(const String & filename, std::vector<AbsoluteQuantitationMethod> & aqm_list);
+
+    /**
+        @brief Stores an AbsoluteQuantitationMethod file.
+
+        @exception Exception::UnableToCreateFile is thrown if the file could not be created
+    */
+    void store(const String & filename, const std::vector<AbsoluteQuantitationMethod> & aqm_list);
+    
+    /**
+        @brief Checks if a file is valid with respect to the mapping file and the controlled vocabulary.
+
+        @param line Header line of the .csv file.
+        @param headers A map of header strings to column positions.
+        @param params_headers A map of transformation model parameter header strings to column positions.
+    */
+    void parseHeader(StringList & line, std::map<String, int> & headers,
```

Comment:
can you make parseHeader and parseLine private?

Reply:
This seems to mess up the tests.  Is there a way of testing private methods?

Reply:
well in principle you should not test private methods. sometimes it is useful though, so you can use a trick: make it protected, in the test inherit from your main class and pass it through. With this we maintain a clean interface and API of the library but can still test private methods

Reply:
Is there an example of this in OpenMS?  I don't quite follow.

Reply:
for example: https://github.com/OpenMS/OpenMS/blob/develop/src/tests/class_tests/openms/source/MasstraceCorrelator_test.cpp#L47

here the function `matchMassTraces_` is `protected` in the parent class but the child class is just a facade that passes through the parameters and it is `public` in the child class. I would suggest something similar.